### DISCUSSION TITLE
Avoid compressing .pdf for transfer

### DISCUSSION THREAD

**Discussion by mikegilchrist** [community/community#22062] [New to GitHub]
<p>I’ve added a bunch of .pdf files to my repo.  It’s slow to upload and, presumably, download.  I assume this is in part due to git trying to compress the already compressed .pdf files.</p>
<ol>
<li>Is my understanding correct? Is git trying to compress .pdf files?</li>
<li>Is there a simple way to avoid this since these files are already compressed?</li>
<li>Why wouldn’t github set itself up to avoid compressing classes of binary files that are generally already compressed?</li>
</ol>
<p>Thanks,</p>
<p>Mike</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/mikegilchrist">@mikegilchrist</a>! :wave: Welcome to the Community!</p>
<p>Git and GitHub are optimized to provide version control and code collaboration predominantly on text files, which means each push of data to our servers triggers computation on our end to apply necessary metadata and structure things efficiently for that purpose.</p>
<p>That means there are many use cases, such as backups of non-text files, or database dumps, that are unsuitable for Git, and an inefficient strain on our infrastructure.</p>
<p>You can read more about this here:</p>
<p><a href="https://help.github.com/en/github/managing-large-files/what-is-my-disk-quota" class="onebox" target="_blank">https://help.github.com/en/github/managing-large-files/what-is-my-disk-quota</a></p>
<p>To answer your questions:</p>
<ol>
<li>
<p>Git isn’t trying to compress your .pdf files in the same way as if you were using an archiving program, exactly - it’s setting up pack files, so that rather storing each revision of your files separately, it stores one version and the computes “deltas” of what has changed in each revision. There’s some interesting discussion about this in <a href="https://www.reddit.com/r/git/comments/1ydvpg/does_git_compress_files_efficiently/">this reddit thread!</a></p>
</li>
<li>
<p>I’m afraid you can’t prevent files from being processed, no.</p>
</li>
<li>
<p>We don’t do this because fundamentally big binaries shouldn’t be committed to the Git history in the first place.</p>
</li>
</ol>
<p>I hope that explains things!</p>

**Comment by mikegilchrist** [comment]
<p>Thanks; that’s helpful.</p>
<p>Note I am not storing any big binary files, I am storing many, many small binary files. Regardless, I will stop worrying about this issue.</p>
